test/s3select_perf_test.cpp

```diff
@@ -0,0 +1,386 @@
+    ss << "9,38439,5645,2.8,Mahalia,Aldric,Mahalia.Aldric@yopmail.com,doctor,2019-04-20T20:21:22.23+05:15,__" << std::endl;
+    ss << "10,6611,7287,1.0,Pamella,Sibyls,Pamella.Sibyls@yopmail.com,police officer,2000-09-13T14:41Z,1_" << std::endl;
+  }
+  out = ss.str();
+}
+
+TEST(TestS3SelectPerformance, parse_with_result)
+{
+  std::string input_csv;
+  generate_data_csv(input_csv, COUNT);
+  std::string input_query = "select _1, _2, _3, _4, _5, _6, _7, _8, _9 from s3object;" ;
+
+  auto start = std::chrono::high_resolution_clock::now();
+
+  ProfilerStart("./parse_with_result_perf.txt");
+
+  auto s3select_res = run_s3select(input_query, input_csv);
+
+  ProfilerFlush();
+  ProfilerStop();
+
+  auto end = std::chrono::high_resolution_clock::now();
+  std::chrono::duration<long, std::micro> elapsed_us =
+          std::chrono::duration_cast<std::chrono::milliseconds>(end - start);
+
+  fprintf(stderr, "Total time taken by query: %" PRIu64 " μs\nAverage time per parse per row: %" PRIu64 " μs\n",
+          elapsed_us.count(), elapsed_us.count() / (COUNT * 10));
+}
+
+TEST(TestS3SelectPerformance, parse_without_result)
+{
+  std::string input_csv;
+  generate_data_csv(input_csv, COUNT);
+  std::string input_query = "select count(*) from s3object where _2 > _3;" ;
+
+  auto start = std::chrono::high_resolution_clock::now();
+
+  ProfilerStart("./parse_without_result_perf.txt");
+
+  auto s3select_res = run_s3select(input_query, input_csv);
+
+  ProfilerFlush();
+  ProfilerStop();
+
+  auto end = std::chrono::high_resolution_clock::now();
+  std::chrono::duration<long, std::micro> elapsed_us =
+          std::chrono::duration_cast<std::chrono::milliseconds>(end - start);
+
+  fprintf(stderr, "Total time taken by query: %" PRIu64 " μs\nAverage time per parse per row: %" PRIu64 " μs\n",
+          elapsed_us.count(), elapsed_us.count() / (COUNT * 10));
+}
+
+TEST(TestS3SelectPerformance, like_dynamic)
+{
+  std::string input_csv;
+  generate_data_csv(input_csv, COUNT);
+  std::string input_query = "select count(*) from s3object where _1 like _10;" ;
+
+  auto start = std::chrono::high_resolution_clock::now();
+
+  ProfilerStart("./parse_without_result_perf.txt");
+
+  auto s3select_res = run_s3select(input_query, input_csv);
+
+  ProfilerFlush();
+  ProfilerStop();
+
+  EXPECT_EQ(s3select_res, "100000,");
+
+  auto end = std::chrono::high_resolution_clock::now();
+  std::chrono::duration<long, std::micro> elapsed_us =
+          std::chrono::duration_cast<std::chrono::milliseconds>(end - start);
+
+  fprintf(stderr, "Total time taken by query: %" PRIu64 " μs\nAverage time per parse per row: %" PRIu64 " μs\n",
+          elapsed_us.count(), elapsed_us.count() / (COUNT * 10));
+}
+
+TEST(TestS3SelectPerformance, like_constant)
+{
+  std::string input_csv;
+  generate_data_csv(input_csv, COUNT);
+  std::string input_query = "select count(*) from s3object where _1 like \"1_\";" ;
```

Comment:
the regular expression is too short.
it is not putting enough stress on the regex-engine